towhee/functional/mixins/milvus_mixin.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2021 Zilliz. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from towhee.hparam import param_scope
+from ..entity import Entity
+
+
+def insert(self, index, collection, batch_size):
+    # pylint: disable=import-outside-toplevel
+    from towhee.utils.milvus_utils import Collection, MutationResult
+    if isinstance(collection, str):
+        collection = Collection(collection)
+    primary_keys = []
+    insert_count = 0
+    data = [[] for _ in range(len(index))]
+
+    for entity in self:
+        for i in range(len(index)):
+            data[i].append(getattr(entity, index[i]))
+        if len(data[0]) == batch_size:
+            mr = collection.insert(data)
+            data = [[] for _ in range(len(index))]
+            primary_keys += mr.primary_keys
+            insert_count += mr.insert_count
+            print(f'Successfully inserted {mr.insert_count} row data.')
+
+    if len(data[0]) > 0:
+        mr = collection.insert(data)
+        primary_keys += mr.primary_keys
+        insert_count += mr.insert_count
+        print(f'Successfully inserted {mr.insert_count} row data.')
+
+    e = Entity(insert_count=insert_count,
+               primary_keys=primary_keys,
+               delete_count=mr.delete_count,
+               upsert_count=mr.upsert_count,
+               timestamp=mr.timestamp)
+    milvus_mr = MutationResult(e)
+    collection.load()
```

Comment:
what if something goes wrong, and we execute `collection.insert` without `collection.load`

Reply:
So finally call `collection.load` with `try-exception`.